gguf-py/gguf/huggingface_hub.py

```diff
@@ -0,0 +1,237 @@
+import json
+import logging
+import os
+import pathlib
+from enum import IntEnum, auto
+from hashlib import sha256
+
+import requests
+from transformers import AutoTokenizer
+
+
+class TokenizerType(IntEnum):
+    SPM = auto()  # SentencePiece
+    BPE = auto()  # BytePair
+    WPM = auto()  # WordPiece
+
+
+class ModelType(IntEnum):
+    PTH = auto()  # PyTorch
+    SFT = auto()  # SafeTensor
+
+
+# NOTE:
+#   - Repository paths are required
+#   - Allow the user to specify the tokenizer model type themselves
+#   - Use architecture types because they are explicitly defined
+#   - Possible tokenizer model types are: SentencePiece, WordPiece, or BytePair
+MODELS = (
+    {"tokt": TokenizerType.SPM, "repo": "meta-llama/Llama-2-7b-hf", },
+    {"tokt": TokenizerType.BPE, "repo": "meta-llama/Meta-Llama-3-8B", },
+    {"tokt": TokenizerType.SPM, "repo": "microsoft/Phi-3-mini-4k-instruct", },
+    {"tokt": TokenizerType.BPE, "repo": "deepseek-ai/deepseek-llm-7b-base", },
+    {"tokt": TokenizerType.BPE, "repo": "deepseek-ai/deepseek-coder-6.7b-base", },
+    {"tokt": TokenizerType.BPE, "repo": "tiiuae/falcon-7b", },
+    {"tokt": TokenizerType.WPM, "repo": "BAAI/bge-small-en-v1.5", },
+    {"tokt": TokenizerType.BPE, "repo": "mosaicml/mpt-7b", },
+    {"tokt": TokenizerType.BPE, "repo": "bigcode/starcoder2-3b", },
+    {"tokt": TokenizerType.BPE, "repo": "openai-community/gpt2", },
+    {"tokt": TokenizerType.BPE, "repo": "smallcloudai/Refact-1_6-base", },
+    {"tokt": TokenizerType.BPE, "repo": "CohereForAI/c4ai-command-r-v01", },
+    {"tokt": TokenizerType.BPE, "repo": "Qwen/Qwen1.5-7B", },
+    {"tokt": TokenizerType.BPE, "repo": "allenai/OLMo-1.7-7B-hf", },
+    {"tokt": TokenizerType.BPE, "repo": "databricks/dbrx-base", },
+    {"tokt": TokenizerType.WPM, "repo": "jinaai/jina-embeddings-v2-base-en", }, # WPM!
+    {"tokt": TokenizerType.BPE, "repo": "jinaai/jina-embeddings-v2-base-es", },
+    {"tokt": TokenizerType.BPE, "repo": "jinaai/jina-embeddings-v2-base-de", },
+    {"tokt": TokenizerType.BPE, "repo": "microsoft/phi-1", },
+    {"tokt": TokenizerType.BPE, "repo": "stabilityai/stablelm-2-zephyr-1_6b", },
+    {"tokt": TokenizerType.SPM, "repo": "mistralai/Mistral-7B-Instruct-v0.2", },
+    {"tokt": TokenizerType.SPM, "repo": "mistralai/Mixtral-8x7B-Instruct-v0.1", },
+)
+
+
+class HFHubRequest:
+    def __init__(self, auth_token: None | str, logger: None | logging.Logger):
+        # Set headers if authentication is available
+        if auth_token is None:
+            self._headers = None
+        else:
+            self._headers = {"Authorization": f"Bearer {auth_token}"}
+
+        # Set the logger
+        if logger is None:
+            logging.basicConfig(level=logging.DEBUG)
+            logger = logging.getLogger("huggingface-hub")
```

Comment:
Not sure what you're going for here?
- Why change/set level to `DEBUG`
- Perhaps use `__name__` instead of `"huggingface-hub"`

Reply:
What would you recommend for the logger level?

Reply:
The caller has most likely already called `basicConfig` in which case this will be a no-op, however if that's not the case the saner default would be `INFO`.